soh/soh/Enhancements/randomizer/3drando/location_access/locacc_zoras_domain.cpp

```diff
@@ -43,7 +43,12 @@ void RegionTable_Init_ZorasDomain() {
                   Entrance(RR_ZR_FAIRY_GROTTO,     {[]{return Here(RR_ZORAS_RIVER, []{return logic->BlastOrSmash();});}}),
                   Entrance(RR_THE_LOST_WOODS,      {[]{return logic->HasItem(RG_SILVER_SCALE) || logic->CanUse(RG_IRON_BOOTS);}}),
                   Entrance(RR_ZR_STORMS_GROTTO,    {[]{return logic->CanOpenStormsGrotto();}}),
-                  Entrance(RR_ZR_BEHIND_WATERFALL, {[]{return logic->CanUse(RG_ZELDAS_LULLABY) || (logic->IsChild && ctx->GetTrickOption(RT_ZR_CUCCO)) || (logic->IsAdult && logic->CanUse(RG_HOVER_BOOTS) && ctx->GetTrickOption(RT_ZR_HOVERS));}}),
+                  Entrance(RR_ZR_BEHIND_WATERFALL, {[]{
+                       return Here(RR_ZORAS_RIVER, []{return ctx->GetOption(RSK_SLEEPING_WATERFALL).Is(RO_WATERFALL_OPEN);}) ||
+                              logic->CanUse(RG_ZELDAS_LULLABY) ||
+                              (logic->IsChild && ctx->GetTrickOption(RT_ZR_CUCCO)) ||
+                              (logic->IsAdult && logic->CanUse(RG_HOVER_BOOTS) && ctx->GetTrickOption(RT_ZR_HOVERS));
+                  }}),
```

Comment:
Keep it, I say. When you guys return to the OI well, you'll need it.

Reply:
Yeah, if it's not going to cause issues to leave it in, I see no reason to get rid of it when we'll need it later, but it should probably be commented out with a note about why it'll be needed later.

Reply:
Make sure it's labelled with a TODO.

Reply:
Actually, I have a dumb question for @Pepper0ni. Why was the Here check needed in the first place? OI would ostensibly only apply if the waterfall is Closed (forcing Once for the enhancement), but the condition is that the rando setting for the waterfall is Open (forcing Never).

Reply:
What should be happening (there seems to be a bug that you pointed out) is that if Once is forced, opening the waterfall as one age would open it as the other. Here to it's own location is a hacky way to check if you can do something as any age (making this less obtuse is on my todo list), so it would check if it was possible to open the waterfall as 1 age to let the other in.

This only matters with OI as both ages can use the ocarina if they have on, but it's possible for you to only have an age-locked item to OI with.

Reply:
I think I wrapped the wrong part of the condition in Here. It should have been `Here(RR_ZORAS_RIVER, []{return logic->CanUse(RG_ZELDAS_LULLABY);})` because you originally said that the ZL check should be what allows either age access. That's what's important to OI when Once is forced.